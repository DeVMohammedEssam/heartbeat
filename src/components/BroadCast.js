import React, { Component } from 'react';
import Navbar from "./layout/Navbar";
import {sendNotification} from "../redux/actions/index"
class BroadCast extends Component {
    state = {
        sendNowClicked: false
    }
    handleSubmit = (e) => {
        e.preventDefault();
        this.setState(() => ({ sendNowClicked: true }));
 /*
        this.props.dispatch(sendNotification(data)).then(()=>{
 this.setState({sendNowClicked:false})
        })
*/
    }
    render() {
        return (
            <React.Fragment>
                <Navbar />
                <form className="broadcast" onSubmit={this.handleSubmit}>
                    <h3 className="broadcast__title">Broadcast</h3>
                    <div className="broadcast__content">
                        <div className="broadcast__textbox">
                            <textarea className="custom-input" name="broadcast-message" placeholder="Send message"></textarea>
                            {this.state.sendNowClicked && (
                                <span className="done-image-container"></span>
                            )}
                        </div>
                        <div className="broadcast__send-to ">
                            <div className="wrapper">

                                <select name="sentTo" className=" custom-btn btn--darkGrey">
                                    <option value="patients">patients</option>
                                    <option value="doctors">negative people</option>
                                </select>
                                <span className="subtitle">message will be sent to: 26,381 users</span>
                            </div>
                            <input className="broadcast__send-now custom-input " type="submit" value="Send Now" />
                        </div>
                    </div>
                </form>
            </React.Fragment>
        );
    }
}

export default BroadCast;