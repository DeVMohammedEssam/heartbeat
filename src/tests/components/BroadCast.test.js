import { shallow } from 'enzyme';
import React from 'react';
import BroadCast from '../../components/BroadCast';
test('should render BroadCast component correctly before submitting the form', () => {
	const wrapper = shallow(<BroadCast />);
	expect(wrapper).toMatchSnapshot();
});
test('should render BroadCast component correctly after submitting the form', () => {
	const wrapper = shallow(<BroadCast />);
	expect(wrapper.state('sendNowClicked')).toBe(false);
	/*     wrapper.simulate("submit", {
            preventDefault: () => { }
        })
         */
	expect(wrapper).toMatchSnapshot();
});
test('should change state of message on textarea change', () => {
	const wrapper = shallow(<BroadCast />);
	const value = 'a message to be broadcasted';
	wrapper.find('textarea').simulate('change', {
		target: { value }
	});
	expect(wrapper.state('broadcast')['message']).toBe(value);
});
/* test("should change send to on select box ", () => {
    const wrapper = shallow(<BroadCast />);
    const select = wrapper.find("select");
    let value = "doctor";
    select.simulate("change", {
        target: { value }
    })
    const state = wrapper.state("broadcast");
    expect(state.sendTo).toBe(value)
}) */
