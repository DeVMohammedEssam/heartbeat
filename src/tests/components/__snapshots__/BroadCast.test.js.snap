// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BroadCast component correctly before submitting the form 1`] = `
<Fragment>
  <ConnectFunction />
  <form
    className="broadcast"
    onSubmit={[Function]}
  >
    <h3
      className="broadcast__title"
    >
      Broadcast
    </h3>
    <div
      className="broadcast__content"
    >
      <div
        className="broadcast__textbox"
      >
        <textarea
          className="custom-input"
          name="message"
          onChange={[Function]}
          placeholder="Send message"
          value=""
        />
      </div>
      <div
        className="broadcast__send-to "
      >
        <div
          className="wrapper"
        >
          <select
            className=" custom-btn btn--darkGrey"
            name="result"
            onChange={[Function]}
          >
            <option
              selected={true}
              value="1"
            >
              patients
            </option>
            <option
              value="0"
            >
              negative people
            </option>
          </select>
          <span
            className="subtitle"
          >
            message will be sent to: 26,381 users
          </span>
        </div>
        <input
          className="broadcast__send-now custom-input "
          type="submit"
          value="Send Now"
        />
      </div>
    </div>
  </form>
</Fragment>
`;
